src/javascript/ContentEditor/editorTabs/AdvancedOptions/VanityUrls/VanityUrls.jsx

```diff
@@ -1,23 +1,38 @@
-import React from 'react';
+import React, {useContext} from 'react';
 import {useTranslation} from 'react-i18next';
 import styles from './VanityUrls.scss';
 import {getButtonRenderer} from '~/ContentEditor/utils';
 import {DisplayAction} from '@jahia/ui-extender';
 import {Typography} from '@jahia/moonstone';
 import clsx from 'clsx';
+import {useNodeChecks} from '@jahia/data-helper';
+import {useContentEditorContext} from '~/ContentEditor/contexts';
 
 const ButtonRendererAccent = getButtonRenderer({defaultButtonProps: {variant: 'default', size: 'big', color: 'accent'}});
 
 export const VanityUrls = () => {
     const {t} = useTranslation('jcontent');
 
+    const {path} = useContentEditorContext();
+
+    const res = useNodeChecks(
+        {path},
+        {
+            requiredPermission: ['viewVanityUrlModal']
+        }
+    );
+
+    const hasPermission = res.checksResult;
+
```

Comment:
You can also use the loading property from useNodeChecks to avoid to display the **noPermissionLabel** before displaying the other label
```

if (loading) {
        /* eslint-disable react/jsx-no-useless-fragment */
        return (<></>);
    }
```

Reply:
Thanks done